File: cpp/jolly_jumpers.cpp
#include <vector>
#include <iostream>
#include <cmath>
using namespace std;

int main() {
    int n;
    while (cin >> n) {
        vector<bool> visited(n, false);
        int prev;
        cin >> prev;
        for (int i = 0; i < n - 1; i++) {
            int curr;
            cin >> curr;
            int diff = abs(curr - prev);
            if (diff < n) {
                visited[diff] = true;
            }
            prev = curr;
        }
        
        bool isJolly = true;
        for (int i = 1; i < n; i++) {
            if (visited[i] == false) {
                isJolly = false;
            }
        }
        cout << (isJolly ? "Jolly" : "Not jolly") << endl;
    }
}

File: java/src/JollyJumpers.java
import java.util.*;

public class JollyJumpers {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        while (input.hasNext()) {
            int n = input.nextInt();
            boolean[] visited = new boolean[n];

            int prev = input.nextInt();
            for (int i = 0; i < n - 1; i++) {
                int curr = input.nextInt();
                int diff = Math.abs(curr - prev);
                if (diff < n) {
                    visited[diff] = true;
                }
                prev = curr;
            }

            boolean isJolly = true;
            for (int i = 1; i < n; i++) {
                if (visited[i] == false) {
                    isJolly = false;
                }
            }
            System.out.println(isJolly ? "Jolly" : "Not jolly");
        }
    }
}
